File: Relative sort array/Relative_sort_array.cpp
class Solution {
public:
    vector<int> relativeSortArray(vector<int>& arr1, vector<int>& arr2) {
        vector<int> res;
        vector<int> temp;
        unordered_map<int, int> countMap;
        for (int num : arr1) {
            countMap[num]++;
        }
        for (int num : arr2) {
            while (countMap[num] > 0) {
                res.push_back(num);
                countMap[num]--;
            }
        }
        for (int num : arr1) {
            if (countMap[num] > 0) {
                temp.push_back(num);
                countMap[num]--;
            }
        }
        sort(temp.begin(), temp.end());
        res.insert(res.end(), temp.begin(), temp.end());

        return res;
    }
};

File: Relative sort array/Relative_sort_array.py
class Solution:
    def relativeSortArray(self, arr1: List[int], arr2: List[int]) -> List[int]:
        n = len(arr1)
        m = len(arr2)
        res = []
        s = set()


        for i in range(m):
            for j in range(n):
                if arr1[j]==arr2[i]:
                    res.append(arr2[i])
                    s.add(arr2[i])
        res1 = []
        for i in arr1 : 
            if i not in s : 
                res1.append(i)

        res = res+sorted(res1)
                    

        return res 
            

        